### PR TITLE
Support Decimal256 on AVG aggregate expression

### LINKED ISSUE #7852: Support Decimal256 on `AVG` aggregate expression
### Is your feature request related to a problem or challenge?

Currently a query containing `AVG` aggregate expression with Decimal256 gets `This feature is not implemented: AvgAccumulator for \(Decimal256\(5, 2\) --> Decimal256\(9, 6\)\)` error.

This ticket is open for Decimal256  support on `AVG` aggregate expression.

### Describe the solution you'd like

Support Decimal256 on `AVG` aggregate expression.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by viirya**
## Which issue does this PR close?

Closes #7852.

### CODE REVIEW

**Review by Dandandan — APPROVED** [review]

File: datafusion/physical-expr/src/aggregate/average.rs [hunk 1]
```diff
@@ -256,40 +292,55 @@ impl Accumulator for AvgAccumulator {
...
     sum_precision: u8,
     target_precision: u8,
     target_scale: i8,
 }
 
-impl Accumulator for DecimalAvgAccumulator {
+impl<T: DecimalType + ArrowNumericType> Debug for DecimalAvgAccumulator<T> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("DecimalAvgAccumulator")
+            .field("sum", &self.sum)
```
Comment (alamb) [inline datafusion/physical-expr/src/aggregate/average.rs:307]: is this needed because `#derive` doesn't know how to handle `Option<T::Native>`?

**Review by alamb — APPROVED** [review]
Looks very nice to me -- thank you @viirya

File: datafusion/physical-expr/src/aggregate/average.rs (hunk 1 above)
Comment (viirya) [reply datafusion/physical-expr/src/aggregate/average.rs:307]: Yea. #derive doesn't work.

**Comment by viirya** [comment]
Thank you @Dandandan @alamb

**Pull request merged**